crates/client/rpc-core/src/lib.rs

```diff
@@ -36,11 +41,25 @@ pub struct PredeployedAccountWithBalance {
     pub balance: FieldElement,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct DeclareV0Result {
+    pub class_hash: ClassHash,
+}
+
 /// Madara rpc interface for additional features.
 #[rpc(server, namespace = "madara")]
 pub trait MadaraRpcApi: StarknetReadRpcApi {
     #[method(name = "predeployedAccounts")]
     fn predeployed_accounts(&self) -> RpcResult<Vec<PredeployedAccountWithBalance>>;
+
+    #[method(name = "declarev0")]
+    async fn declare_v0_contract(
```

Comment:
let's add a comment for the issue we were facing when sending the class info directly